sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/SecretProperties.java

```diff
@@ -71,33 +72,42 @@
      * include: 'Purgeable', 'Recoverable+Purgeable', 'Recoverable',
      * 'Recoverable+ProtectedSubscription'.
      */
-    private String recoveryLevel;
+    String recoveryLevel;
 
     /**
      * The content type of the secret.
      */
     @JsonProperty(value = "contentType")
-    private String contentType;
+    String contentType;
 
     /**
      * Application specific metadata in the form of key-value pairs.
      */
     @JsonProperty(value = "tags")
-    private Map<String, String> tags;
+    Map<String, String> tags;
 
     /**
      * If this is a secret backing a KV certificate, then this field specifies
      * the corresponding key backing the KV certificate.
      */
     @JsonProperty(value = "kid", access = JsonProperty.Access.WRITE_ONLY)
-    private String keyId;
+    String keyId;
 
     /**
      * True if the secret's lifetime is managed by key vault. If this is a
      * secret backing a certificate, then managed will be true.
      */
     @JsonProperty(value = "managed", access = JsonProperty.Access.WRITE_ONLY)
-    private Boolean managed;
+    Boolean managed;
+
+    SecretProperties(String secretName) {
+        this.name = secretName;
```

Comment:
shouldn't this have a require non-null?

Reply:
this is validated by the service.
As per guidelines, not needed.